### PR TITLE
New rule: `new-color-css-vars`

### PR DESCRIPTION
**Pull request by langermank**
Adds a new rule to update old color CSS variables with new variables, including a fallback value. This should only run on the `sx` prop as we have Stylelint in place to cover the `styled` syntax.

### CODE REVIEW

File: src/rules/__tests__/new-color-css-vars.test.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      code: `<Blankslate border></Blankslate>`
+    }
+  ],
+  invalid: [
+    {
+      code: `<Button sx={{color: 'var(--color-fg-muted)'}}>Test</Button>`,
+      output: `<Button sx={{color: 'var(--fgColor-muted, var(--color-fg-muted))'}}>Test</Button>`,
+      errors: [
+        {
+          message: 'Replace var(--color-fg-muted) with var(--fgColor-muted, var(--color-fg-muted))'
```
Comment (joshblack) [inline src/rules/__tests__/new-color-css-vars.test.js:38]: Will folks always need to include a fallback value or is this an interim step as we're moving over to the new system?

**Review by joshblack — APPROVED** [review]
Looks great! Just had a question

File: src/rules/__tests__/new-color-css-vars.test.js (hunk 1 above)
Comment (langermank) [reply src/rules/__tests__/new-color-css-vars.test.js:38]: Interim step! Once v8 is released and fully rolled out, we will refactor this to not include the fallback. We could also add some logic to remove the old fallbacks, but they won't really hurt anything if they stick around.

**Review by TylerJDev — APPROVED** [review]
Love it 🤩!

One tiny comment, do we need a docs page for this, or nah? (e.g. [`no-deprecated-colors`](https://github.com/primer/eslint-plugin-primer-react/blob/main/docs/rules/no-deprecated-colors.md))

**Review by broccolinisoup — APPROVED** [review]
Looks great! 🤩

I think we should add a documentation and include the rule in the README as well 🤗  

- For docs: https://github.com/primer/eslint-plugin-primer-react/tree/main/docs/rules
- For README: https://github.com/primer/eslint-plugin-primer-react/blob/main/README.md#rules

**Comment by langermank** [comment]
@broccolinisoup @TylerJDev going to merge this and add the docs in a separate PR!

**Pull request merged**